src/features/d3tree/reducers.js

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable no-param-reassign */
+import { createSlice } from "@reduxjs/toolkit";
+
+const initialState = {
+  widthSpacing: 150,
+  heightSpacing: 250,
+  layoutWidth: 400,
+  layoutHeight: 850,
+};
+
+const d3treeSlice = createSlice({
+  name: "d3tree",
+  initialState,
+  reducers: {
+    setLayout(state, action) {
+      state.layoutWidth = action.payload.clientWidth;
+      state.layoutHeight = action.payload.clientHeight;
+    },
+    setWidthSpacing(state, action) {
+      state.widthSpacing = action.payload.width;
+    },
+    setHeightSpacing(state, action) {
+      state.heightSpacing = action.payload.height;
+    },
+  },
+});
+
+export const d3treeActions = d3treeSlice.actions;
```

Comment:
```suggestion
export const { setLayout, setWidthSpacing, setHeightSpacing } = d3treeSlice.actions;
```
구조분해할당으로 export 하시고, 나중에 다른 컴포넌트에서 import할 때는 예를들어 `import { setLayout } from "../features/<<경로>>";` 이렇게 하셔도 될 것 같습니다.